download-basket/src/main/java/downloadbasket/actions/DownloadInfo.java

```diff
@@ -0,0 +1,48 @@
+package downloadbasket.actions;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionHandler;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.map.layer.OskariLayerServiceIbatisImpl;
+import fi.nls.oskari.util.ResponseHelper;
+import fi.nls.oskari.util.JSONHelper;
+import downloadbasket.helpers.SendDownloadDetailsToEmailThread;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+/**
+ * Checks all download- and user details of the download basket when starting
+ * the download process. Returns boolean "success".
+ */
+
+@OskariActionRoute("DownloadInfo")
+public class DownloadInfo extends ActionHandler {
+
+	private final Logger LOGGER = LogFactory.getLogger(DownloadInfo.class);
+
+	private static final String PARAM_DOWNLOAD_DETAILS = "downloadDetails";
+	private static final String PARAM_USER_DETAILS = "userDetails";
+	OskariLayerService mapLayerService;
+
+	@Override
+	public void handleAction(final ActionParameters params) throws ActionException {
+
+		String downloadDetails = params.getRequiredParam(PARAM_DOWNLOAD_DETAILS);
+		String strUserDetails = params.getRequiredParam(PARAM_USER_DETAILS);
+		try {
+			JSONObject userDetails = new JSONObject(strUserDetails);
+			JSONArray ddArray = new JSONArray(downloadDetails);
+			mapLayerService = new OskariLayerServiceIbatisImpl();
```

Comment:
This should be initialized only once for this ActionHandler in an overridden version of the init() function (https://github.com/oskariorg/oskari-server/blob/develop/service-control/src/main/java/fi/nls/oskari/control/ActionHandler.java#L31-L36)